odl/solvers/functional/default_functionals.py

```diff
@@ -59,18 +59,18 @@ class LpNorm(Functional):
         \| x \|_p = \\left(\\int_\Omega |x(t)|^p dt. \\right)^{1/p}
     """
 
-    def __init__(self, space, exponent):
+    def __init__(self, domain, exponent, range=None):
         """Initialize a new instance.
 
         Parameters
         ----------
-        space : `DiscreteLp` or `TensorSpace`
+        domain : `DiscreteLp` or `TensorSpace`
             Domain of the functional.
         exponent : float
             Exponent for the norm (``p``).
         """
         super(LpNorm, self).__init__(
-            space=space, linear=False, grad_lipschitz=np.nan)
+            domain=domain, linear=False, grad_lipschitz=np.nan, range=range)
```

Comment:
So this is step 1: allow to manually set the range. I'm wondering about step 2: the sensible default for norms and such. What about
```python
if range is None and getattr(domain, 'is_complex', False):
    range = RealNumbers()
super(...)
```